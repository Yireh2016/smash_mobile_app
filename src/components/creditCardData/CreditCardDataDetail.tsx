import * as React from 'react';
import { View, StyleSheet } from 'react-native';
import TableRow from '../tableRow/TableRow';
import { Card } from '../../types/Card';
import { P1, H3 } from '../texts/Texts';
import CardListItem from '../cardListItem/CardListItem';
import dateFormats from '../../helpers/dateFormats/dateFormats';
import formatMoneyValue from '../../helpers/formatMoneyValue/formatMoneyValue';
import formatPercentage from '../../helpers/formatPercentage/formatPercentage';

interface CreditCardDataDetailProps {
  card: Card;
}

const styles = StyleSheet.create({
  tableRowCont: {
    marginTop: 17,
  },
  cardPropContainer: {
    padding: 18,
  },
});
const CreditCardDataDetail: React.FunctionComponent<CreditCardDataDetailProps> = ({
  card,
}) => {
  const {
    current_balance,
    credit_utilization_percentage,
    statement_balance,
    purchase_apr_percentage,
    min_payment,
    payment_due,
  } = card;

  const cardPropertiesArr = [
    `Current balance:${formatMoneyValue(current_balance)}`,
    `Credit utilization:${formatPercentage(credit_utilization_percentage)}`,
    `Statement balance:${formatMoneyValue(statement_balance)}`,
    `Purchase APR:${formatPercentage(purchase_apr_percentage)}`,
    `Minimum payment:${formatMoneyValue(min_payment)}`,
    `Payment due date:${dateFormats(payment_due)}`,
  ];

  const cardPropertiesArrMap = cardPropertiesArr.map(
    (cardProperties: string, index) => {
      const [key, value] = cardProperties.split(':');
      return (
        <View key={index} style={index ? styles.tableRowCont : {}}>
          <TableRow
            columns={[
              {
                component: <P1>{key}</P1>,
                align: 'left',
              },
              {
                component: <H3>{value}</H3>,
                align: 'right',
              },
            ]}
          />
        </View>
      );
    }
  );
  return (
    <CardListItem>
      <View style={styles.cardPropContainer}>{cardPropertiesArrMap}</View>
    </CardListItem>
  );
};

export default CreditCardDataDetail;
