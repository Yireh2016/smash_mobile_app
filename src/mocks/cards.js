export default [
  {
    account_id: '2u3ywjhedb',
    bank: 'Citi',
    number: '2267',
    current_balance: 500,
    credit_utilization_percentage: 10,
    statement_balance: 2650,
    purchase_apr_percentage: 19,
    min_payment: 65,
    payment_due: '2020-12-18',
    other_aprs: [
      {
        apr_type: 'purchase_apr',
        apr_percentage: 19,
        balance_subject_to_apr: 0,
      },
      {
        apr_type: 'cash_apr',
        apr_percentage: 27.95,
        balance_subject_to_apr: 0,
      },
      {
        apr_type: 'adjustment',
        apr_percentage: 0,
        balance_subject_to_apr: 3100,
      },
    ],
  },
  {
    account_id: '2u3ywjhedb',
    bank: 'Citi',
    number: '9376',
    current_balance: 2000,
    credit_utilization_percentage: 25,
    statement_balance: 1030,
    purchase_apr_percentage: 21,
    min_payment: 37,
    payment_due: '2020-12-18',
    other_aprs: [
      {
        apr_type: 'purchase_apr',
        apr_percentage: 21,
        balance_subject_to_apr: 0,
      },
      {
        apr_type: 'cash_apr',
        apr_percentage: 27.95,
        balance_subject_to_apr: 0,
      },
      {
        apr_type: 'adjustment',
        apr_percentage: 0,
        balance_subject_to_apr: 1100,
      },
    ],
  },
  {
    account_id: '2u3ywjhedb',
    bank: 'Citi',
    number: '9085',
    current_balance: 3000,
    credit_utilization_percentage: 10,
    statement_balance: 1820,
    purchase_apr_percentage: 22,
    min_payment: 42,
    payment_due: '2020-12-18',
    other_aprs: [
      {
        apr_type: 'purchase_apr',
        apr_percentage: 22,
        balance_subject_to_apr: 0,
      },
      {
        apr_type: 'cash_apr',
        apr_percentage: 27.95,
        balance_subject_to_apr: 0,
      },
      {
        apr_type: 'adjustment',
        apr_percentage: 0,
        balance_subject_to_apr: 1900,
      },
    ],
  },
];
