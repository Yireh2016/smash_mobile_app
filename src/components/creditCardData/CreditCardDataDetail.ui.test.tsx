import * as React from 'react';
import renderer from 'react-test-renderer';
import CreditCardDataDetail from './CreditCardDataDetail';
import cards from '../../mocks/cards';
it('CreditCardDataDetail renders correctly', () => {
  const ComponentProps = {
    card: cards[0],
  };
  const Element = () => {
    return <CreditCardDataDetail {...ComponentProps} />;
  };

  const rendered = renderer.create(<Element />).toJSON();
  expect(rendered).toMatchSnapshot();
});
