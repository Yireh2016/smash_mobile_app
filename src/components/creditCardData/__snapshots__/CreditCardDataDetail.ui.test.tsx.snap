// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreditCardDataDetail renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#142836",
      "borderRadius": 12,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "padding": 18,
      }
    }
  >
    <View
      style={Object {}}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "width": "50%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontFamily": "OpenSans-Regular",
                "fontSize": 16,
              }
            }
          >
            Current balance
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "width": "50%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontFamily": "OpenSans-Bold",
                "fontSize": 18,
              }
            }
          >
            $500
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "marginTop": 17,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "width": "50%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontFamily": "OpenSans-Regular",
                "fontSize": 16,
              }
            }
          >
            Credit utilization
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "width": "50%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontFamily": "OpenSans-Bold",
                "fontSize": 18,
              }
            }
          >
            10%
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "marginTop": 17,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "width": "50%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontFamily": "OpenSans-Regular",
                "fontSize": 16,
              }
            }
          >
            Statement balance
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "width": "50%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontFamily": "OpenSans-Bold",
                "fontSize": 18,
              }
            }
          >
            $2,650
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "marginTop": 17,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "width": "50%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontFamily": "OpenSans-Regular",
                "fontSize": 16,
              }
            }
          >
            Purchase APR
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "width": "50%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontFamily": "OpenSans-Bold",
                "fontSize": 18,
              }
            }
          >
            19%
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "marginTop": 17,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "width": "50%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontFamily": "OpenSans-Regular",
                "fontSize": 16,
              }
            }
          >
            Minimum payment
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "width": "50%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontFamily": "OpenSans-Bold",
                "fontSize": 18,
              }
            }
          >
            $65
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "marginTop": 17,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "width": "50%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontFamily": "OpenSans-Regular",
                "fontSize": 16,
              }
            }
          >
            Payment due date
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "width": "50%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontFamily": "OpenSans-Bold",
                "fontSize": 18,
              }
            }
          >
            12/18
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
